src/daft-plan/src/optimization/rules/push_down_projection.rs

```diff
@@ -401,3 +488,222 @@ impl OptimizerRule for PushDownProjection {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::sync::Arc;
+
+    use common_error::DaftResult;
+    use daft_core::{datatypes::Field, DataType};
+    use daft_dsl::{binary_op, col, lit, Operator};
+
+    use crate::{
+        optimization::{
+            optimizer::{RuleBatch, RuleExecutionStrategy},
+            rules::PushDownProjection,
+            Optimizer,
+        },
+        test::dummy_scan_node,
+        LogicalPlan,
+    };
+
+    /// Helper that creates an optimizer with the PushDownProjection rule registered, optimizes
+    /// the provided plan with said optimizer, and compares the optimized plan's repr with
+    /// the provided expected repr.
+    fn assert_optimized_plan_eq(plan: Arc<LogicalPlan>, expected: &str) -> DaftResult<()> {
+        let optimizer = Optimizer::with_rule_batches(
+            vec![RuleBatch::new(
+                vec![Box::new(PushDownProjection::new())],
+                RuleExecutionStrategy::Once,
+            )],
+            Default::default(),
+        );
+        let optimized_plan = optimizer
+            .optimize_with_rules(
+                optimizer.rule_batches[0].rules.as_slice(),
+                plan.clone(),
+                &optimizer.rule_batches[0].order,
+            )?
+            .unwrap()
+            .clone();
+        assert_eq!(optimized_plan.repr_indent(), expected);
+
+        Ok(())
+    }
+
+    /// Projection merging: Ensure factored projections do not get merged.
+    #[test]
+    fn test_merge_does_not_unfactor() -> DaftResult<()> {
+        let a2 = binary_op(Operator::Plus, &col("a"), &col("a"));
```

Comment:
Nit: I believe that `col("a") + col("a")` and similar below should work since the arithmetic operators or implemented for `Expr`, right?

```suggestion
        let a2 = col("a") + col("a");
```